src/legacy/core_plugins/kibana/public/management/sections/objects/components/objects_table/components/flyout/flyout.js

```diff
@@ -99,24 +109,259 @@ class FlyoutUI extends Component {
+        }),
+      });
+      return;
+    }
+
+    // Go through the failures and split between unmatchedReferences and failedImports
+    const failedImports = [];
+    const unmatchedReferences = new Map();
+    for (const { error, ...obj } of response.errors || []) {
+      failedImports.push({ obj, error });
+      if (error.type !== 'missing_references') {
+        continue;
+      }
+      // Currently only supports resolving references on index patterns
+      const indexPatternRefs = error.references.filter(ref => ref.type === 'index-pattern');
+      for (const missingReference of indexPatternRefs) {
+        const conflict = unmatchedReferences.get(`${missingReference.type}:${missingReference.id}`) || {
+          existingIndexPatternId: missingReference.id,
+          list: [],
+          newIndexPatternId: undefined,
+        };
+        conflict.list.push(obj);
+        unmatchedReferences.set(`${missingReference.type}:${missingReference.id}`, conflict);
+      }
+    }
+
+    this.setState({
+      failedImports,
+      unmatchedReferences: Array.from(unmatchedReferences.values()),
+      isLoading: false,
+      importCount: response.successCount,
+      wasImportSuccessful: unmatchedReferences.size === 0 && !failedImports.some(issue => issue.error && issue.error.type === 'conflict'),
+      conflictedSavedObjectsLinkedToSavedSearches: undefined,
+      conflictedSearchDocs: undefined,
+      possibleRecordsToOverwrite: (response.errors || []).filter(obj => obj.error.type === 'conflict'),
+    });
+
+    // Resolve import errors right away if there's no index patterns to match
+    // This will ask about overwriting each object, etc
+    if (unmatchedReferences.size === 0) {
+      this.resolveImportErrors();
+    }
+  }
+
+  /**
+   * Get Conflict Resolutions
+   *
+   * Function iterates through the objects, displays a modal for each asking the user if they wish to overwrite it or not.
+   *
+   * @param {array} objects List of objects to request the user if they wish to overwrite it
+   * @return {Promise<array>} An object with the key being "type:id" and value the resolution chosen by the user
+   */
+  async getConflictResolutions(objects) {
+    const resolutions = {};
+    for (const { type, id, title } of objects) {
+      const overwrite = await new Promise((resolve) => {
+        this.setState({
+          conflictingRecord: {
+            id,
+            type,
+            title,
+            done: resolve,
+          },
+        });
+      });
+      resolutions[`${type}:${id}`] = overwrite;
+      this.setState({ conflictingRecord: undefined });
+    }
+    return resolutions;
+  }
+
+  /**
+   * Map Import Failure To Retry Object
+   *
+   * Takes a single import failure and builds a retry object for the resolve import errors API. Returns undefined when failure should be skipped.
+   */
+  mapImportFailureToRetryObject = ({ failure, overwriteDecisionCache, replaceReferencesCache }) => {
```

Comment:
This part is definitely complex, but I understand why. There's a lot of state being tracked.

I wonder if the complexity of tracking these references and conflicts could be made simpler if the state was represented as a graph (which this problems seems to map to very naturally) rather than this cache that has to be passed around? Would take some experimentation to see if it fits better, but could be worth the exploration.

Reply:
As discussed, we will address this when the flyout component gets refactored and legacy code is removed.